### PR TITLE
CI: run automatically only for master and PRs, add workflow_dispatch for manual CI runs

### PR DESCRIPTION
**Pull request by benlorenz**
We still get quite some backlog, so I think it would be better to run the tests only for master + PRs and not for the branches. This should free up some of the runners, especially for macos.
Anyone who needs to run the tests on his branch can either create a Draft PR or run them manually with the [workflow_dispatch](https://docs.github.com/en/actions/managing-workflow-runs/manually-running-a-workflow).

cc: @thofma @fingolfin 

PS: This PR should only trigger the tests once instead of twice.

### CODE REVIEW

**Comment by thofma** [comment]
Thanks. Seems to work :)

**Review by thofma — APPROVED** [review]

**Comment by benlorenz** [comment]
The workflow dispatch button should appear [here](https://github.com/oscar-system/Oscar.jl/actions/workflows/CI.yml) but this might not happen until the workflow has run once (or maybe even only after the merge?).

**Comment by thofma** [comment]
Looking at the queue, I guess we will find out tomorrow.

**Comment by benlorenz** [comment]
Another speedup possibility would be to enable `fail-fast` (except for julia nightly probably) ?
To avoid running through many tests even if those for one julia version or OS already failed, but sometimes the output of different runs might still be helpful...

I think this would work with setting `fail-fast: true` and adding:
```
    continue-on-error: ${{ matrix.julia-version == 'nightly' }}
```

**Comment by thofma** [comment]
Not sure about that one. I would suggest we merge this here and see if this is already enough. But we should keep that option in mind.

**Comment by benlorenz** [comment]
> Not sure about that one. I would suggest we merge this here and see if this is already enough. But we should keep that option in mind.

agreed

**Pull request merged**
